cs/CMakeLists.txt

```
@@ -118,6 +118,7 @@ set(BASE_SOURCES
   ${BASE_DIR}/printer.cpp
   ${BASE_DIR}/release_string.c
   ${BASE_DIR}/resource_tracker.cpp
+  ${BASE_DIR}/server_type.cpp
```

Also remove server_type.cpp
```suggestion
  ${BASE_DIR}/resource_tracker.cpp
```